enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/Steps.java

```diff
@@ -0,0 +1,662 @@
+            return compartments.size() == 1;
+        }
+    }
+
+    /**
+     * Context of SuggestedStep. Determines next step.
+     */
+    public interface SuggestedContext {
+
+        String getItemType();
+
+        Step getNextStep();
+
+        public void setItemType(String selected);
+    }
+
+    /**
+     * Show list of items for a suggested type.
+     * 
+     */
+    static class SuggestedStep implements Step<CompartmentItem, OCIItem> {
+
+        private Map<String, OCIItem> items = new HashMap<>();
+        private OCIItem selected;
+        private Lookup lookup;
+        private SuggestedContext context = null;
+
+        public SuggestedStep prepare(CompartmentItem compartment, Lookup lookup) {
+            this.lookup = lookup;
+            context = lookup.lookup(SuggestedContext.class);
+            ProgressHandle h = ProgressHandle.createHandle(Bundle.CollectingItems());
+            h.start();
+            h.progress(Bundle.CollectingItems_Text());
+            try {
+                getItemsByPath(compartment, context.getItemType()).forEach((db) -> items.put(db.getName(), db));
+            } finally {
+                h.finish();
+            }
+            return this;
+        }
+
+        private String getSuggestedItemName() {
+            switch (context.getItemType()) {
+                case "Databases":
+                    return "Oracle Autonomous Database";
+                case "Vault":
+                    return "OCI Vault";
+                case "Bucket":
+                    return "Bucket";
+                case "Cluster":
+                    return "Cluster";
+                case "ComputeInstance":
+                    return "ComputeInstance";
+                default:
+                    return "X";
+            }
+        }
+
+        @Override
+        public NotifyDescriptor createInput() {
+            return createQuickPick(items, Bundle.SelectItem(getSuggestedItemName()));
+        }
+
+        @Override
+        public Step getNext() {
+            Step next = context.getNextStep();
+            if (next != null) {
+                next.prepare(getValue(), lookup);
+            }
+            return next;
+        }
+
+        @Override
+        public void setValue(String selected) {
+            this.selected = items.get(selected);
+        }
+
+        @Override
+        public OCIItem getValue() {
+            if (onlyOneChoice()) {
+                selected = items.values().iterator().next();
+            }
+            return selected;
+        }
+
+        @Override
+        public boolean onlyOneChoice() {
+            return false;
+        }
+    }
+
+    private static <T extends OCIItem> NotifyDescriptor.QuickPick createQuickPick(Map<String, T> ociItems, String title) {
+        List<NotifyDescriptor.QuickPick.Item> items = ociItems.entrySet().stream()
+                .map(entry -> new NotifyDescriptor.QuickPick.Item(entry.getKey(), entry.getValue().getDescription()))
```

Comment:
It seems that description is filed for just some OCIItems; could it default to the item's name if description is empy/null ?